sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/GenerateOrdering.scala

```diff
@@ -193,14 +193,30 @@ object GenerateOrdering extends CodeGenerator[Seq[SortOrder], Ordering[InternalR
 /**
  * A lazily generated row ordering comparator.
  */
-class LazilyGeneratedOrdering(val ordering: Seq[SortOrder])
+class LazilyGeneratedOrdering(
+    val ordering: Seq[SortOrder], secondOrdering: => Seq[SortOrder] = Seq.empty)
   extends Ordering[InternalRow] with KryoSerializable {
 
   def this(ordering: Seq[SortOrder], inputSchema: Seq[Attribute]) =
     this(ordering.map(BindReferences.bindReference(_, inputSchema)))
 
   @transient
-  private[this] var generatedOrdering = GenerateOrdering.generate(ordering)
+  private[this] var generatedOrdering = {
+    var generatedOrdering = GenerateOrdering.generate(ordering)
+    if (!secondOrdering.isEmpty) {
+      secondOrdering.foreach { order =>
```

Comment:
These two lines are redundant

Reply:
Logic here used to filter input order which can not generate ordering normally.